Add userguide information to docs (#3391)
MDAnalysis/mdanalysis: package/doc/sphinx/source/index.rst

> .. _installation-instructions: 

Reviewer: Put this link target in front of installation.
```suggestion
```

User Guide
==========

> The MDAnalysis `userguide`_ provides comprehensive information on how to

Reviewer: I'd write it *User Guide* consistently, at least that's how https://userguide.mdanalysis.org/stable/index.html refers to itself.
```suggestion
The MDAnalysis `User Guide`_ provides comprehensive information on how to
```

use the library. We would recommend that new users have a look at the
`quick start guide`_. The userguide also has a set of `examples`_ on how to
use the MDAnalysis library which may be of interest.

> .. _`userguide`: https://userguide.mdanalysis.org/stable/index.html

Reviewer: ```suggestion
.. _`User Guide`: https://userguide.mdanalysis.org/stable/index.html
```

.. _`quick start guide`: https://userguide.mdanalysis.org/stable/examples/quickstart.html
.. _`examples`: https://userguide.mdanalysis.org/stable/examples/README.html

> 

Reviewer: Moved link target
```suggestion

.. _installation-instructions: 
```

**Source code** is available from
https://github.com/MDAnalysis/mdanalysis/ under the `GNU Public
Licence, version 2`_. Obtain the sources with `git`_.

.. code-block:: bash

   git clone https://github.com/MDAnalysis/mdanalysis.git

> The userguide provides more information on how to

Reviewer: ```suggestion
The `User Guide`_ provides more information on how to
```
